src/core/Akka.API.Tests/CoreAPISpec.ApproveCore.approved.txt

```diff
@@ -2351,86 +2343,72 @@ namespace Akka.Dispatch
         Akka.Actor.Settings Settings { get; }
     }
     public interface IMultipleConsumerSemantics : Akka.Dispatch.ISemantics { }
+    public interface IProducesMessageQueue<TQueue>
+        where TQueue : Akka.Dispatch.MessageQueues.IMessageQueue { }
     public interface IRequiresMessageQueue<T>
         where T : Akka.Dispatch.ISemantics { }
+    public interface IRunnable
+    {
+        void Run();
+    }
     public interface ISemantics { }
     public interface IUnboundedDequeBasedMessageQueueSemantics : Akka.Dispatch.IDequeBasedMessageQueueSemantics, Akka.Dispatch.ISemantics, Akka.Dispatch.IUnboundedMessageQueueSemantics { }
     public interface IUnboundedMessageQueueSemantics : Akka.Dispatch.ISemantics { }
-    public abstract class Mailbox : System.IDisposable
-    {
-        protected Akka.Dispatch.MessageDispatcher dispatcher;
-        protected System.Boolean modreq(System.Runtime.CompilerServices.IsVolatile) hasUnscheduledMessages;
-        protected int status;
-        protected Mailbox() { }
-        protected Akka.Actor.ActorCell ActorCell { get; }
-        public abstract bool IsClosed { get; }
-        public bool IsSuspended { get; }
-        public abstract void BecomeClosed();
-        public abstract void CleanUp();
+    public class Mailbox
```

Comment:
The new `Mailbox` type - **breaking changes here**.
 
This type is now firmly aligned with the JVM's design, and in implementing this I remove methods that were not relevant to that design. Many properties that were on the previous implementation are now callable methods instead, so we could take advantage of `AggressiveInling` for key state checking operations.

All mailbox instances are more or less the same, the major difference being the `IMessageQueue` instance produced by the `MailboxType` - which will be injected into this class at the time of construction.